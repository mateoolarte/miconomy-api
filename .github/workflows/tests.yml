name: CI tests

on: push

jobs:
  tests:
    name: Test API
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Install dependencies
        run: yarn --frozen-lockfile
      - name: Run tests
        env:
          DATABASE_URL: ${{ secrets.DATABASE_URL }}
          APP_SECRET: ${{ secrets.APP_SECRET }}
        run: yarn test
  build:
  runs-on: ubuntu-latest
  strategy:
    matrix:
      node-version: '15.x'
  steps:
    - uses: actions/checkout@v2
    - name: Use Node.js
      uses: actions/setup-node@v1
