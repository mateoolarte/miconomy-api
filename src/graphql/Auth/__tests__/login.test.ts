import { createTestContext } from '../../../tests/__helpers';

const ctx = createTestContext();

const signupMutation = `
  mutation signup($email: String!, $password: String!) {
    signup(email: $email, password: $password) {
      user {
        id
        email
        currencyCode
      }
    }
  }
`;

const loginMutation = `
  mutation login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      user {
        id
        email
        currencyCode
      }
    }
  }
`;

describe('login', () => {
  it('should give access to a existing user', async () => {
    const signupUser = await ctx.client.request(signupMutation, {
      email: 'test2@miconomy.co',
      password: '12345678',
    });

    expect(signupUser).toMatchSnapshot();

    const loginUser = await ctx.client.request(loginMutation, {
      email: 'test2@miconomy.co',
      password: '12345678',
    });

    expect(loginUser).toMatchSnapshot();
  });
});
