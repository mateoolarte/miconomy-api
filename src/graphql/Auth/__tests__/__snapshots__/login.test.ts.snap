// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should give access to a existing user 1`] = `
Object {
  "signup": Object {
    "user": Object {
      "currencyCode": "COP",
      "email": "test2@miconomy.co",
      "id": 1,
    },
  },
}
`;

exports[`login should give access to a existing user 2`] = `
Object {
  "login": Object {
    "user": Object {
      "currencyCode": "COP",
      "email": "test2@miconomy.co",
      "id": 1,
    },
  },
}
`;
